pkg/engine/executor/executor_test.go

```diff
@@ -97,3 +104,127 @@ func TestExecutor_Projection(t *testing.T) {
 		require.ErrorContains(t, err, "projection expects exactly one input, got 2")
 	})
 }
+
+func TestExecutor_Parse(t *testing.T) {
+	t.Run("no inputs result in empty pipeline", func(t *testing.T) {
+		ctx := t.Context()
+		c := &Context{}
+		pipeline := c.executeParse(ctx, &physical.ParseNode{
+			Kind:          logical.ParserLogfmt,
+			RequestedKeys: []string{"level", "status"},
+		}, nil)
+		err := pipeline.Read(ctx)
+		require.ErrorContains(t, err, EOF.Error())
+	})
+
+	t.Run("multiple inputs result in error", func(t *testing.T) {
+		ctx := t.Context()
+		c := &Context{}
+		pipeline := c.executeParse(ctx, &physical.ParseNode{
+			Kind:          logical.ParserLogfmt,
+			RequestedKeys: []string{"level"},
+		}, []Pipeline{emptyPipeline(), emptyPipeline()})
+		err := pipeline.Read(ctx)
+		require.ErrorContains(t, err, "parse expects exactly one input, got 2")
+	})
+}
+
+func TestNewParsePipeline(t *testing.T) {
+	for _, tt := range []struct {
+		name                 string
+		schema               *arrow.Schema
+		input                arrowtest.Rows
+		expectedFields       int
+		expectedParsedOutput map[string][]string
+	}{
+		{
+			name: "parse stage transforms records, adding columns parsed from message",
+			schema: arrow.NewSchema([]arrow.Field{
+				{Name: types.ColumnNameBuiltinMessage, Type: arrow.BinaryTypes.String},
+			}, nil),
+			input: arrowtest.Rows{
+				{"message": "level=error status=500"},
+				{"message": "level=info status=200"},
+				{"message": "level=debug status=201"},
+			},
+			expectedFields: 3, // 3 columns: message, level, status
+			expectedParsedOutput: map[string][]string{
+				"level":  {"error", "info", "debug"},
+				"status": {"500", "200", "201"},
+			},
+		},
+		{
+			name: "parse stage preserves existing columns",
+			schema: arrow.NewSchema([]arrow.Field{
+				{Name: types.ColumnNameBuiltinTimestamp, Type: arrow.FixedWidthTypes.Timestamp_ns},
+				{Name: types.ColumnNameBuiltinMessage, Type: arrow.BinaryTypes.String},
+				{Name: "app", Type: arrow.BinaryTypes.String},
+			}, nil),
+			input: arrowtest.Rows{
+				{"timestamp": time.Unix(1, 0).UTC(), "message": "level=error status=500", "app": "frontend"},
+				{"timestamp": time.Unix(2, 0).UTC(), "message": "level=info status=200", "app": "backend"},
+			},
+			expectedFields: 5, // 5 columns: timestamp, message, app, level, status
+			expectedParsedOutput: map[string][]string{
+				"level":  {"error", "info"},
+				"status": {"500", "200"},
+			},
```

Comment:
I'd recommend changing these to use `arrowtest.Rows`:

```go
expectedParsedOutput: arrowtest.Rows{
  {"level": "error", "status": "500"}, 
  {"level: "info", "status": "200"}, 
}
```

then you can assert using:

```go
record, _ := pipeline.Value() 
defer record.Release()

actual, err := arrowtest.RecordRows(record)
require.NoError(t, err) 
require.Equal(t, tt.expectedParseOutput, actual) 
```

That would avoid the need for the getColumnIndices/verifyParseOutput helper functions, and generally row-order expctations are easier to read than column-order.